presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1391,7 +1392,7 @@ public PhysicalOperation visitIndexJoin(IndexJoinNode node, LocalExecutionPlanCo
         public PhysicalOperation visitJoin(JoinNode node, LocalExecutionPlanContext context)
         {
             List<JoinNode.EquiJoinClause> clauses = node.getCriteria();
-            if (clauses.isEmpty() && !node.getFilter().isPresent()) {
+            if (clauses.isEmpty() && !node.getFilter().isPresent() && node.getType() == INNER) {
```

Comment:
Yeah. But I would rather have it in separate PR.
Handling OUTER JOIN when build side is OUTER side is not trivial. I can mimic how it was done for lookup joins but this is somewhat complicated.

I do not want this to be part of this PR, which is supposed to be bugfix PR.